src/app/features/login/login.component.html

```diff
@@ -1,24 +1,26 @@
 <div class="border border-primary app-login">
     <h2 class="text-center">Login</h2>
-    <form form="ngForm" (ngSubmit)="login()">
+    <form name="form" #ngform="ngForm" (ngSubmit)="ngform.form.valid && login()">
```

Comment:
Adding logical operators for boolean inputs is fine, but here it looks not good. Lets move all logic into login method. Or you can disable submit button based on form status.